cpp/tests/join/conditional_join_tests.cu

```diff
@@ -127,6 +125,26 @@ gen_random_nullable_repeated_columns(unsigned int N = 10000, unsigned int num_re
                    std::pair(std::move(right), std::move(right_nulls)));
 }
 
+// Basic trivially copyable replacement for thrust::pair for use with `device_uvector`
```

Comment:
What's wrong with `thrust::pair`? Can you document it?

Reply:
It's not trivially copyable because it defines non-trivial move/copy constructors.

Reply:
I've added a longer description of this.